server/api/crud/artifacts.py

```diff
@@ -271,6 +271,11 @@ def _delete_artifact_data(
                 producer_id=producer_id,
                 object_uid=object_uid,
             )
+            artifact_kind = artifact["kind"]
+            if artifact_kind in ["model", "dataset", "dir"]:
+                raise mlrun.errors.MLRunNotImplementServerError(
+                    f"Deleting artifact data kind: {artifact_kind} is currently not supported"
+                )
```

Comment:
Please add a comment explaining why we do this now, and a `TODO` to remove once we add support.